pylti/flask.py

```diff
@@ -352,11 +358,15 @@ def wrapper(*args, **kwargs):
                 return error(exception=exception)
 
         return wrapper
-    lti_kwargs_out['app'] = app
-    lti_kwargs_out['request'] = request
-    lti_kwargs_out['error'] = error
-    lti_kwargs_out['role'] = role
-
-    ret = partial(_lti, lti_args=lti_args_out, lti_kwargs=lti_kwargs_out)
 
-    return ret
+    lti_kwargs['request'] = request
+    lti_kwargs['error'] = error
+    lti_kwargs['role'] = role
+
+    if (not app) or isinstance(app, Flask):
+        lti_kwargs['app'] = app
+        return _lti
+    else:
+        # We are wrapping without arguments
+        lti_kwargs['app'] = None
+        return _lti(app)
```

Comment:
I changed this quite a bit after the merge conflict.  It was a long strange trip, but I really wanted to just use `@lti` instead of `@lti()`.  I also didn't understand why he was using a partial, so I removed that.